lib/fetchers/PortionFetcher.ts

```diff
@@ -0,0 +1,105 @@
+import { Unit } from 'aws-embedded-metrics';
+import NodeCache from 'node-cache';
+import axios from '../providers/quoters/helpers';
+import { log } from '../util/log';
+import { metrics } from '../util/metrics';
+
+export enum PortionType {
+  Flat = 'flat',
+  Regressive = 'regressive',
+}
+
+export interface Portion {
+  readonly bips: number;
+  readonly receiver: string;
+  readonly type: PortionType;
+}
+
+export interface GetPortionResponse {
+  readonly hasPortion: true;
+  readonly portion: Portion;
+}
+
+export interface GetNoPortionResponse {
+  readonly hasPortion: false;
+}
```

Comment:
Condense into 
{
hasPortion: bool;
portion?: Portion;
}

Reply:
And I'd prefer the response to not contain `hasPortion` and rather use definedness checks instead for the two states

Reply:
cant we just return null when we dont have it vs true/false for the rest?

Reply:
I agree with just having a single `GetPortionResponse` type with `hasPortion` as boolean, I don't think we need the union, do we?